libraries/botbuilder-dialogs-adaptive/src/entityAssignment.ts

```diff
@@ -5,9 +5,10 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License.
  */
+import { isEqual } from 'lodash';
 import { EntityInfo } from './entityInfo';
 
-export interface EntityAssignment {
+export interface EntityAssignmentConfiguration {
```

Comment:
Do we need an interface that's separate from the class?

Reply:
I did this so that we could deserialize `Partial` JSON objects into the class. I considered a constructor that took all arguments as optional, but that would've been 8 arguments and it seemed like a typed object might be better? I also needed to ensure that any properties not defined were explicitly set `undefined` as explained [here](https://github.com/microsoft/botbuilder-js/pull/3628#discussion_r621496959).

I went back and forth a bit about how best to do this and can't say I figured out the "right" solution. Happy to adjust if you think I should.

Reply:
Makes sense - let's keep brainstorming ways to express "an instance JSON parsed from a string that has extra computed properties" as it seems like a common pattern, and is something that isn't provided out of the box like it is in .NET. For now, this works.